tst/EnergyPlus/unit/UtilityRoutines.unit.cc

```diff
@@ -208,3 +208,52 @@ TEST_F(EnergyPlusFixture, UtilityRoutines_appendPerfLog2)
     // clean up the file
     fs::remove(state->dataStrGlobals->outputPerfLogFilePath);
 }
+
+TEST_F(EnergyPlusFixture, UtilityRoutines_ProcessNumber)
+{
+    // acceptable strings
+    std::string goodString{"3.14159"};
+    double expectedVal{3.14159};
+    bool expectedError{false};
+    EXPECT_NEAR(UtilityRoutines::ProcessNumber(goodString, expectedError), expectedVal, 1E-5);
+    EXPECT_EQ(expectedError, false);
+
+    goodString = "3.14159+E0";
+    EXPECT_NEAR(UtilityRoutines::ProcessNumber(goodString, expectedError), expectedVal, 1E-5);
+    EXPECT_EQ(expectedError, false);
+
+    goodString = "3.14159+e0";
+    EXPECT_NEAR(UtilityRoutines::ProcessNumber(goodString, expectedError), expectedVal, 1E-5);
+    EXPECT_EQ(expectedError, false);
+
+    goodString = "3.14159+D0";
+    EXPECT_NEAR(UtilityRoutines::ProcessNumber(goodString, expectedError), expectedVal, 1E-5);
+    EXPECT_EQ(expectedError, false);
+
+    goodString = "3.14159+d0";
+    EXPECT_NEAR(UtilityRoutines::ProcessNumber(goodString, expectedError), expectedVal, 1E-5);
+    EXPECT_EQ(expectedError, false);
+
+    // invalid strings
+    std::string badString{"É.14159"};
+    expectedVal = 0.0;
+    EXPECT_NEAR(UtilityRoutines::ProcessNumber(badString, expectedError), expectedVal, 1E-5);
+    EXPECT_EQ(expectedError, true);
+
+    badString = "3.14159É0";
+    expectedVal = 0.0;
+    EXPECT_NEAR(UtilityRoutines::ProcessNumber(badString, expectedError), expectedVal, 1E-5);
+    EXPECT_EQ(expectedError, true);
+
+    badString = "3.14159 0";
+    expectedVal = 0.0;
+    EXPECT_NEAR(UtilityRoutines::ProcessNumber(badString, expectedError), expectedVal, 1E-5);
+    EXPECT_EQ(expectedError, true);
+
+    // out of range
+    badString = "1E5000";
```

Comment:
This is a bad string, not out of range. Out of range would be1.0+e5000. And shouldn't you be using EXPECT_FALSE(expectedError) or EXPECT_TRUE(expectedError)?

Also note line 259 below.

Reply:
We're definitely falling into the ```out_of_range``` ``` catch``` block here. And yes, I should be using ```EXPECT_TRUE``` and ```EXPECT_FALSE```. I'll wait for more comments before fixing that.

Reply:
OK then, just wanted to be sure.

Reply:
I don't see any non-valid characters in this test. As far as I can see that would wrap this up.

        if (PString.find_first_not_of(ValidNumerics) == std::string::npos) {

Reply:
I guess the space is non-valid so maybe your OK.

   static std::string const ValidNumerics("0123456789.+-EeDd");

Reply:
We have this "É" in multiple places.

https://github.com/NREL/EnergyPlus/pull/8819/files#diff-9a44f76d9c12fb11c779204ce7954252f59a856841b113185db69cf62ad3ae56R238